core/src/avm2/object.rs

```diff
@@ -881,15 +586,45 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
         multiname: &Multiname<'gc>,
         args: &[Value<'gc>],
         activation: &mut Activation<'_, 'gc>,
-    ) -> Result<Object<'gc>, Error<'gc>> {
+    ) -> Result<Value<'gc>, Error<'gc>> {
```

Comment:
I don't think the primitive construction process is correct- Booleans should default to false if constructed without arguments, and I don't think Number/int/uint/String should do a (potentially-failing) coercion.

Reply:
This logic was moved from the instance initializer for each primitive class - all of our tests are still passing